### PR TITLE
Rescue MissingAttributeError on attribute writing

### LINKED ISSUE #156: Migrations break on devise user/active admin model
I am encountering an issue when trying to enumerize an attribute on a devise User model, specifically the one generated by Active Admin.

### Steps to reproduce

* Create new Rails app
* Install Devise and Active Admin
* Add an attribute (e.g. `role`) to the `AdminUsers` table via migration
* `enumerize` said attribute in `AdminUsers` model
* Run migrations

I have built a sample rails app that exhibits this exact problem: https://github.com/padde/enumerize-bugreport

### Error

During `rake db:migrate` the following error pops up – it seems as if Enumerize is trying to set a default value for a not-yet-existing attribute:

```
ActiveModel::MissingAttributeError: can't write unknown attribute `role'
```

Full stack trace:

```
$ rake db:migrate --trace
** Invoke db:migrate (first_time)
** Invoke environment (first_time)
** Execute environment
** Invoke db:load_config (first_time)
** Execute db:load_config
** Execute db:migrate
== 20140604154620 DeviseCreateAdminUsers: migrating ===========================
-- create_table(:admin_users)
   -> 0.0050s
-- add_index(:admin_users, :email, {:unique=>true})
   -> 0.0011s
-- add_index(:admin_users, :reset_password_token, {:unique=>true})
   -> 0.0005s
== 20140604154620 DeviseCreateAdminUsers: migrated (0.0067s) ==================

rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

can't write unknown attribute `role'/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/attribute_methods/write.rb:72:in `write_attribute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/attribute_methods/dirty.rb:68:in `write_attribute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/activerecord.rb:51:in `write_attribute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/attribute.rb:95:in `role='
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/base.rb:96:in `public_send'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/base.rb:96:in `block in _set_default_value_for_enumerized_attributes'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/attribute_map.rb:23:in `block in each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/attribute_map.rb:22:in `each_pair'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/attribute_map.rb:22:in `each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/base.rb:80:in `_set_default_value_for_enumerized_attributes'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:424:in `block in make_lambda'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:221:in `call'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:221:in `block in halting_and_conditional'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:86:in `call'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:86:in `run_callbacks'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/core.rb:201:in `initialize'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/base.rb:48:in `initialize'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/inheritance.rb:30:in `new'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/inheritance.rb:30:in `new'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/validations.rb:39:in `create!'
$HOME/Desktop/enumerize-bugreport/db/migrate/20140604154620_devise_create_admin_users.rb:5:in `migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:752:in `migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:992:in `block in execute_migration_in_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:1038:in `block in ddl_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/connection_adapters/abstract/database_statements.rb:211:in `block in transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/connection_adapters/abstract/database_statements.rb:219:in `within_new_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/connection_adapters/abstract/database_statements.rb:211:in `transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/transactions.rb:208:in `transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:1038:in `ddl_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:991:in `execute_migration_in_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:953:in `block in migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:949:in `each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:949:in `migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:807:in `up'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:785:in `migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/railties/databases.rake:34:in `block (2 levels) in <top (required)>'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:240:in `call'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:240:in `block in execute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:235:in `each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:235:in `execute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:165:in `invoke'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:150:in `invoke_task'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `block in top_level'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:115:in `run_with_threads'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:100:in `top_level'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:78:in `block in run'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:176:in `standard_exception_handling'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:75:in `run'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/bin/rake:33:in `<top (required)>'
/usr/local/opt/rbenv/versions/2.1.1/bin/rake:23:in `load'
/usr/local/opt/rbenv/versions/2.1.1/bin/rake:23:in `<main>'
ActiveModel::MissingAttributeError: can't write unknown attribute `role'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/attribute_methods/write.rb:72:in `write_attribute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/attribute_methods/dirty.rb:68:in `write_attribute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/activerecord.rb:51:in `write_attribute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/attribute.rb:95:in `role='
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/base.rb:96:in `public_send'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/base.rb:96:in `block in _set_default_value_for_enumerized_attributes'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/attribute_map.rb:23:in `block in each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/attribute_map.rb:22:in `each_pair'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/attribute_map.rb:22:in `each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/base.rb:80:in `_set_default_value_for_enumerized_attributes'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:424:in `block in make_lambda'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:221:in `call'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:221:in `block in halting_and_conditional'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:86:in `call'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activesupport-4.1.1/lib/active_support/callbacks.rb:86:in `run_callbacks'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/core.rb:201:in `initialize'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/enumerize-0.8.0/lib/enumerize/base.rb:48:in `initialize'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/inheritance.rb:30:in `new'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/inheritance.rb:30:in `new'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/validations.rb:39:in `create!'
$HOME/Desktop/enumerize-bugreport/db/migrate/20140604154620_devise_create_admin_users.rb:5:in `migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:752:in `migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:992:in `block in execute_migration_in_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:1038:in `block in ddl_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/connection_adapters/abstract/database_statements.rb:211:in `block in transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/connection_adapters/abstract/database_statements.rb:219:in `within_new_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/connection_adapters/abstract/database_statements.rb:211:in `transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/transactions.rb:208:in `transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:1038:in `ddl_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:991:in `execute_migration_in_transaction'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:953:in `block in migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:949:in `each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:949:in `migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:807:in `up'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/migration.rb:785:in `migrate'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/activerecord-4.1.1/lib/active_record/railties/databases.rake:34:in `block (2 levels) in <top (required)>'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:240:in `call'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:240:in `block in execute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:235:in `each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:235:in `execute'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:179:in `block in invoke_with_call_chain'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/2.1.0/monitor.rb:211:in `mon_synchronize'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:172:in `invoke_with_call_chain'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/task.rb:165:in `invoke'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:150:in `invoke_task'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `block (2 levels) in top_level'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `each'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:106:in `block in top_level'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:115:in `run_with_threads'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:100:in `top_level'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:78:in `block in run'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:176:in `standard_exception_handling'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/lib/rake/application.rb:75:in `run'
/usr/local/opt/rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rake-10.3.2/bin/rake:33:in `<top (required)>'
/usr/local/opt/rbenv/versions/2.1.1/bin/rake:23:in `load'
/usr/local/opt/rbenv/versions/2.1.1/bin/rake:23:in `<main>'
Tasks: TOP => db:migrate
```

### Ugly Workaround

For now, I could fix the issue by creating the following initializer:

```ruby
# config/initializers/enumerize.rb

module Enumerize
  module ActiveRecordSupport
    alias_method :enumerize_without_guard_clause, :enumerize

    def enumerize(name, options={})
      return unless (columns_hash.has_key?(name.to_s) rescue false)
      enumerize_without_guard_clause(name, options)
    end
  end
end
```

Is this a bug or am I just doing it wrong? Thanks!

### PR DESCRIPTION
**Pull request by embs**
Closes #156.

### CODE REVIEW

File: test/activerecord_test.rb [hunk 1]
```diff
@@ -53,6 +53,8 @@ class User < ActiveRecord::Base
   enumerize :status, :in => { active: 1, blocked: 2 }, scope: true
 
   enumerize :account_type, :in => [:basic, :premium]
+
+  enumerize :relationship, :in => [:single, :married]
```
Comment (lest) [inline test/activerecord_test.rb]: Can you please leave a comment that there is no such column in the migration for purpose?

File: lib/enumerize/activerecord.rb [hunk 2]
```diff
@@ -31,7 +31,10 @@ def write_attribute(attr_name, value)
       def becomes(klass)
         became = super
         klass.enumerized_attributes.each do |attr|
-          became.send("#{attr.name}=", send(attr.name))
+          begin
+            became.send("#{attr.name}=", send(attr.name))
+          rescue ActiveModel::MissingAttributeError
```
Comment (lest) [inline lib/enumerize/activerecord.rb]: A comment why we need to rescue this exception would be helpful.

**Comment by lest** [comment]
@embs Thanks a lot! It would be great to add comments to the code describing the purpose of rescuing exceptions. Would mind adding them?

**Comment by embs** [comment]
Added some comments, @lest.

I saw that some comments include links to the related issues but I think they polute the code. Moreover, the related issues can always be tracked down through the versioning system. I can include the links if you wish, though.

**Comment by lest** [comment]
:+1: Awesome! Thank you!

**Comment by padde** [comment]
:heart: :heart: :heart:

**Pull request merged**